joblib/func_inspect.py

```diff
@@ -169,7 +170,11 @@ def getfullargspec(func):
     try:
         return inspect.getfullargspec(func)
     except AttributeError:
-        arg_spec = inspect.getargspec(func)
+        if 3 > sys.version_info[0]:
+            arg_spec = inspect.getargspec(func)
```

Comment:
inspect.getfullargspec doesn't give a DeprecationWarning for me on Python 3.5 so I would be inclined leaving it like it is for now and revisit this when it does.

Longer story: IIRC scikit-learn has a backport of inspect.signature but this requires collections.OrderedDict so they have a backport of collections.OrderedDict for Python 2.6. I would wait until we drop support for Python 2.6 before looking at getting rid of inspect.getfullargspec. There may be more details in https://github.com/joblib/joblib/pull/283 and issues linked there.